climpred/relative_entropy.py

```diff
@@ -0,0 +1,301 @@
+        curv (bool): if curvilinear grids disables EOF weights.
+        ntime (int): number of timesteps calculated.
+        nmember_control (int): number of members created from
+                               bootstrapping from control
+
+    Returns:
+        rel_ent (xr.Dataset): relative entropy
+
+    """
+    # Defaults
+    if neofs is None:
+        neofs = initialized.member.size
+    if ntime is None:
+        ntime = initialized.time.size
+
+    # case if you submit control with dim time and member, LENS case
+    if 'member' in control.dims:
+        control_uninitialized = _bootstrap_dim(
+            control, initialized.time.size, time_dim='time',
+            dim='initialization',
+            dim_label=list(initialized.initialization.values))
+
+    # case if you only submit control with dim time, PM case
+    else:
+        control_uninitialized = xr.concat([
+            _bootstrap_dim(control,
+                           initialized.time.size, dim='member',
+                           dim_label=np.arange(nmember_control))
+            for _ in range(initialized.initialization.size)],
+            dim='initialization')
+        control_uninitialized['initialization'] = initialized.initialization.values
+
+    # initialized and control_uninitialized are allowed to have different
+    # dims as I need more members to sample my control distr. properly
+    if set(initialized.dims) == set(control_uninitialized.dims):
+        warnings.warn(
+            "Warning: initialized and control_uninitialized have different coords.")
+
+    # the datasets should only contain one variable
+    if isinstance(control_uninitialized, xr.Dataset):
+        control_uninitialized = control_uninitialized.to_array().squeeze()
+    if isinstance(initialized, xr.Dataset):
+        initialized = initialized.to_array().squeeze()
+
+    # detrend
+    non_spatial_dims = set(control_uninitialized.dims).intersection(
+        ['initialization', 'member'])
+    non_spatial_dims = list(non_spatial_dims)
+    if not anomaly_data:  # if ds, control are raw values
+        anom_x = initialized - control_uninitialized.mean(non_spatial_dims)
+        anom_b = control_uninitialized - \
+            control_uninitialized.mean(non_spatial_dims)
+    else:  # leave as is when already anomalies
+        anom_x = initialized
+        anom_b = control_uninitialized
+
+    # prepare for EOF
+    if curv:  # if curvilinear lon(x,y), lat(x,y) data inputs
+        wgts = None
+    else:
+        coslat = np.cos(np.deg2rad(anom_x.coords['lat'].values))
+        wgts = np.sqrt(coslat)[..., np.newaxis]
+
+    if isinstance(control, xr.Dataset):  # EOF requires xr.dataArray
+        control = control.to_array().squeeze()
+
+    if 'member' in control.dims:  # LENS
+        # stack member and initialization into time dim, make time first
+        non_spatial_control_dims = list(
+            set(control.dims).intersection(['time', 'member']))
```

Comment:
This is where it should be `['initialization', 'member']` since 'time' implies it has a lead time dimension